### PR TITLE
JavaScript function, string ,number, float, parameter examples solved.

### PR DESCRIPTION
**Pull request by Ezgi3406**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

File: exercises/F-strings-methods/exercise2.js [hunk 1]
```diff
@@ -1,7 +1,4 @@
...
-var length=name.length;
-console.log(`my name is ${name.toLowerCase()} and my name is ${length}  characters long.`);
-// or
-console.log(`my name is ${name} and my name is ${length}  characters long.`)
-
-
+const name1 = " Daniel  ";
+var length=name1.length;
+console.log(`my name is ${name1.toLowerCase()} and my name is ${length}  characters long.`);
+console.log(`my name is ${name1} and my name is ${length}  characters long.`)
```
Comment (HamraKhan) [inline exercises/F-strings-methods/exercise2.js:4]: Great use of interpolation here.

File: mandatory/4-tax.js [hunk 2]
```diff
@@ -4,9 +4,16 @@
...
 */
-
-function calculateSalesTax() {}
-
+// price*product gives you the total price without tax.
+// 1 + salesTax gives you the price after tax.
+function calculateSalesTax(price,product) {
+  var salesTax = .20;
+  var totalPrice = (price * product) * (1+salesTax);
+  return "£"+totalPrice;
```
Comment (jstadnik) [inline mandatory/4-tax.js:12]: I think there's some confusion what this function is supposed to do. There's no formatting that is supposed to happen here, you're simply to return the number. I.e. 
```suggestion
  return totalPrice;
```

File: mandatory/4-tax.js [hunk 3]
```diff
@@ -4,9 +4,16 @@
...
   Sales tax is 20% of the price of the product.
 */
-
-function calculateSalesTax() {}
-
+// price*product gives you the total price without tax.
+// 1 + salesTax gives you the price after tax.
+function calculateSalesTax(price,product) {
+  var salesTax = .20;
+  var totalPrice = (price * product) * (1+salesTax);
```
Comment (jstadnik) [inline mandatory/4-tax.js:11]: I think all you need here is the "price" alone -- notice how in all your `console.log` tests you passed the price in as `1`, only varying the product. Then you can simply call this function as `calculateSalesTax(15)`.

What was supposed to be the purpose of `price` here? Were you thinking of quantity?

File: mandatory/4-tax.js [hunk 4]
```diff
@@ -17,7 +24,15 @@ function calculateSalesTax() {}
   Remember that the prices must include the sales tax (hint: you already wrote a function for this!)
 */
 
-function addTaxAndFormatCurrency() {}
+function addTaxAndFormatCurrency(num,product) {
+  let tax=0.20;
+  let total=(num*product)*(1+tax);
```
Comment (jstadnik) [inline mandatory/4-tax.js:29]: Once you amend the `calculateSalesTax` function as described above, you can replace this calculation with a call to the `caclulateSalesTax` function.

File: extra/2-piping.js [hunk 5]
```diff
@@ -15,27 +15,37 @@
   3. Write a more readable version of what you wrote in step 2 under the BETTER PRACTICE comment. Assign
   the final result to the variable goodCode
 */
-
-function add() {
-
+function add(a,b) {
```
Comment (jstadnik) [inline extra/2-piping.js:18]: Let's go over this question in a meeting. It seems this one proved particularly hard for everyone.

**Review by jstadnik — COMMENTED** [review]
A good start, well done on attempting some of the extra exercises :slightly_smiling_face: 

You are already familiar with some more advanced javascript, so I struggle sometimes to know whether you made a mistake, or tried to use something cleverer than the question asks you to -- apologies if I got this wrong.

Good use of string interpolation, as someone mentioned!

**Pull request closed without merging**